packages/client/src/Call.ts

```diff
@@ -276,35 +279,51 @@ export class Call {
       ringing ? CallingState.RINGING : CallingState.IDLE,
     );
 
-    this.on('all', (event) => {
-      // update state with the latest event data
-      this.state.updateFromEvent(event);
-    });
-
-    this.leaveCallHooks.add(
-      registerEventHandlers(this, this.state, this.dispatcher),
-    );
-    this.registerEffects();
-
-    this.leaveCallHooks.add(
-      createSubscription(
-        this.trackSubscriptionsSubject.pipe(
-          debounce((v) => timer(v.type)),
-          map((v) => v.data),
-        ),
-        (subscriptions) =>
-          this.sfuClient?.updateSubscriptions(subscriptions).catch((err) => {
-            this.logger('debug', `Failed to update track subscriptions`, err);
-          }),
-      ),
-    );
-
     this.camera = new CameraManager(this);
     this.microphone = new MicrophoneManager(this);
     this.speaker = new SpeakerManager(this);
     this.screenShare = new ScreenShareManager(this);
   }
 
+  private async setup() {
+    await withoutConcurrency(this.joinLeaveConcurrencyTag, async () => {
```

Comment:
Prevents a race condition with `leave`. It's important to wait for `leave` to finish before we set everything back up again. See the test case here: https://github.com/GetStream/stream-video-js/pull/1433/files#diff-912706ed8d680e55d746f9765aa5090d64771196a9894262bed3e0b73d9a8c97R92-R110